packages/@uppy/aws-s3-multipart/types/index.d.ts

```diff
@@ -25,6 +34,8 @@ export interface AwsS3MultipartOptions extends PluginOptions {
       file: UppyFile,
       opts: { uploadId: string; key: string; signal: AbortSignal }
     ) => MaybePromise<AwsS3Part[]>
+    getTemporarySecurityCredentials?: boolean | (() => MaybePromise<AwsS3STSResponse>)
+    getTemporarySecurityCredentialsExpiry?: (credentials: AwsS3STSResponse['credentials'], options?: {signal?: AbortSignal}) => MaybePromise<number>
```

Comment:
we could type this more strictly to only allow `boolean` if `companionUrl` is set. It's a bit more involved but here is an example:

https://github.com/transloadit/uppy/blob/7a90f9b20c0fbc815ecf786cf4bdbc45448d0814/packages/%40uppy/transloadit/types/index.d.ts#L116-L153

What do you think?

Reply:
Good thinking, I think it should go on its own PR, because the other options (e.g. `signPart`, `abortMultipartUpload`, etc.) would also benefit for more fine-grained typings.